From 56d4a508c1e78a266aa7b1990f81c14fe5fa6393 Mon Sep 17 00:00:00 2001
From: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
Date: Mon, 27 Sep 2021 16:29:43 +0200
Subject: [PATCH] fpnew: Fix Vivado flow

Work around the following Vivado error:
[Synth 8-523] part-select [15:0] does not match declaration
[fpnew_sdotp_multi.sv:211]

Signed-off-by: Nils Wistoff <nwistoff@iis.ee.ethz.ch>
---
 src/fpnew_sdotp_multi.sv | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/fpnew_sdotp_multi.sv b/src/fpnew_sdotp_multi.sv
index 19b849b..b490d70 100644
--- a/src/fpnew_sdotp_multi.sv
+++ b/src/fpnew_sdotp_multi.sv
@@ -207,10 +207,12 @@ module fpnew_sdotp_multi #(
   assign dst_fmt_q  = inp_pipe_dst_fmt_q[NUM_INP_REGS];
 
   logic [3:0][SRC_WIDTH-1:0] operands_post_inp_pipe;
-  assign operands_post_inp_pipe[3] = operand_d_q;
-  assign operands_post_inp_pipe[2] = operand_c_q[SRC_WIDTH-1:0];
-  assign operands_post_inp_pipe[1] = operand_b_q;
-  assign operands_post_inp_pipe[0] = operand_a_q[SRC_WIDTH-1:0];
+  for (genvar i = 0; i < SRC_WIDTH; i++) begin : gen_op_assign
+    assign operands_post_inp_pipe[3][i] = operand_d_q[i];
+    assign operands_post_inp_pipe[2][i] = operand_c_q[i];
+    assign operands_post_inp_pipe[1][i] = operand_b_q[i];
+    assign operands_post_inp_pipe[0][i] = operand_a_q[i];
+  end
 
   // -----------------
   // Input processing
-- 
2.16.5

